pkg/cmd/issue/shared/lookup.go

```diff
@@ -72,17 +72,26 @@ func findIssueOrPR(httpClient *http.Client, repo ghrepo.Interface, number int, f
 		}
 	}
 
+	// isPinned is an invalid field for PullRequest
+	prFields := make([]string, len(fields))
+	for i, f := range fields {
+		if f == "isPinned" {
+			prFields = append(fields[:i], fields[i+1:]...)
+			break
+		}
+	}
+
 	query := fmt.Sprintf(`
 	query IssueByNumber($owner: String!, $repo: String!, $number: Int!) {
 		repository(owner: $owner, name: $repo) {
 			hasIssuesEnabled
 			issue: issueOrPullRequest(number: $number) {
 				__typename
 				...on Issue{%[1]s}
-				...on PullRequest{%[1]s}
+				...on PullRequest{%[2]s}
 			}
 		}
-	}`, api.PullRequestGraphQL(fields))
+	}`, api.PullRequestGraphQL(fields), api.PullRequestGraphQL(prFields))
```

Comment:
lmk if there is a better way to do this. My graphql knowledge is limited.

Reply:
[`repository`](https://docs.github.com/en/graphql/reference/objects#repository) has `pinnedIssues` (see [here](https://github.com/rsteube/carapace-bin/blob/master/pkg/actions/tools/gh/issue.go#L39)).
Might potentially give better performance but could also opt for this solution for consistency.